src/main/java/io/nats/client/support/JwtUtils.java

```diff
@@ -0,0 +1,145 @@
+        String jwt = issueUserJWT(accountSeed, issuerAccount, userKeyPub, optionalName, optionalExpiration, optionalTags);
+        return String.format(NATS_USER_JWT_FORMAT, jwt, userSeed);
+    }
+
+    public static String issueUserCreds(NKey accountSigningKey, String issuerAccount, String userSeed, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        String jwt = issueUserJWT(accountSigningKey, issuerAccount, userKeyPub, optionalName, optionalExpiration, optionalTags);
+        return String.format(NATS_USER_JWT_FORMAT, jwt, userSeed);
+    }
+
+    public static String issueUserJWT(String accountSeed, String issuerAccount, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        NKey accountSigningKey = NKey.fromSeed(accountSeed.toCharArray());
+        return issueUserJWT(accountSigningKey, issuerAccount, userKeyPub, optionalName, optionalExpiration, optionalTags);
+    }
+
+    public static String issueUserJWT(NKey accountSigningKey, String issuerAccount, String userKeyPub, String optionalName, Duration optionalExpiration, String... optionalTags) throws GeneralSecurityException, IOException {
+        String accSigningKeyPub = new String(accountSigningKey.getPublicKey());
+
+        Claim claim = new Claim();
+        claim.exp = optionalExpiration;
+        claim.iat = System.currentTimeMillis() / 1000;
+        claim.iss = accSigningKeyPub;
+        claim.name = Validator.nullOrEmpty(optionalName) ? userKeyPub : optionalName;
+        claim.sub = userKeyPub;
+        claim.nats = new Nats();
+        claim.nats.issuerAccount = issuerAccount;
+        claim.nats.tags = optionalTags;
+
+        // Issue At time is stored in unix seconds
+        String claimJson = claim.toJson();
+
+        // Compute jti, a base32 encoded sha256 hash
+        MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
+        byte[] encoded = sha256.digest(claimJson.getBytes(StandardCharsets.UTF_8));
+
+        claim.jti = new String(NKey.base32Encode(encoded));
+        claimJson = claim.toJson();
+
+        // all three components (header/body/signature) are base64url encoded
+        String encBody = ToBase64Url(claimJson);
+
+        // compute the signature off of header + body (. included on purpose)
+        byte[] sig = (ENCODED_CLAIM_HEADER + "." + encBody).getBytes(StandardCharsets.UTF_8);
+        String encSig = ToBase64Url(accountSigningKey.sign(sig));
+
+        // append signature to header and body and return it
+        return ENCODED_CLAIM_HEADER + "." + encBody + "." + encSig;
+    }
+
+    public static String ToBase64Url(byte[] input) {
```

Comment:
I'd suggest leaving these as private and use the more standard `toBase64Url` (lower-case first) convention.

Reply:
I've been in .NET too much lately, yes please fix. Also maybe belongs in the utility with the base32 function.

Reply:
Update to the more standard camelCase.